x/distribution/keeper/allocation.go

```diff
@@ -40,6 +45,32 @@ func (k Keeper) AllocateTokens(
 		return
 	}
 
+	remaining := feesCollected
+
+	// If block contains entropy then use proportion of block reward to reward to validators which successfully
+	// completed the dkg. If block does not contain entropy then this proportion of the fees collected goes into
+	// the community pool
+	beaconRewardMultiplier := k.GetBeaconReward(ctx)
+	vals := []*tmtypes.Validator{}
+	if len(entropy.GroupSignature) != 0 {
+		// Get all validators (including those than have been jailed)
+		k.stakingKeeper.IterateLastValidators(ctx, func(index int64, val exported.ValidatorI) bool {
+			vals = append(vals, tmtypes.NewValidator(val.GetConsPubKey(), val.GetConsensusPower()))
+			return false
+		})
+
+		// Pay beacon rewards to those in qual
+		valSet := tmtypes.NewValidatorSet(vals)
+		for _, index := range entropy.SuccessfulVals {
+			addr, val := valSet.GetByIndex(int(index))
+			validator := k.stakingKeeper.ValidatorByConsAddr(ctx, sdk.ConsAddress(addr))
+			powerFraction := sdk.NewDec(val.VotingPower).QuoTruncate(sdk.NewDec(valSet.TotalVotingPower()))
```

Comment:
Minor: presumably the powerFraction and reward lines are always the same and could be set when remaining is?

Reply:
No, they are always the same. The powerFraction depends  on the validators voting power.